WeakAurasOptions/OptionsFrames/TextEditor.lua

```diff
@@ -92,6 +94,103 @@ local function set_scheme()
   color_scheme["not"] = theme["Logical"]
 end
 
+-- Define the premade snippets
+local premadeSnippets = {
+  {
+    name = "Basic function",
+    snippet = [=[
+function()
+
+    return
+end]=]
+  },
+  {
+    name = "Custom Activation",
+    snippet = [=[
+function(trigger)
+    return trigger[1] and (trigger[2] or trigger[3])
+end]=]
+  },
+  {
+    name = "Trigger: CLEU",
+    snippet = [=[
+function(event, timestamp, subEvent, hideCaster, sourceGUID, sourceName, sourceFlags, sourceRaidFlags, destGUID, destName, destFlags, destRaidFlags, ...)
+
+    return
+end]=]
+  },
+  {
+    name = "Simple throttle",
+    snippet = [=[
+if not aura_env.last or aura_env.last < GetTime() - 1 then
+    aura_env.last = GetTime()
+
+end]=]
+  },
+  {
+    name = "Trigger State Updater",
+    snippet = [=[
+function(allstates, event, ...)
+    allstates[""] = {
+        show = true,
+        changed = true,
+        progressType = "static"||"timed",
+        value = ,
+        total = ,
+        duration = ,
+        expirationTime = ,
+        autoHide = true,
+        name = ,
+        icon = ,
+        stacks = ,
+        index = ,
+    }
+    return true
+end]=]
+  },
+  {
+    name = "Text: Decimals (percentage)",
+    snippet = [=[
+function()
+    -- Change percentpower as needed
+    -- Change [1] to other your trigger number
+    -- The 0 in `"%.0f"` controls how many decimal places it will round to
+    if aura_env.states[1] and aura_env.states[1].percentpower then
+        return string.format("%.0f", aura_env.states[1].percentpower)
+    end
+end]=]
+  },
+  {
+    name = "Text: Abbreviate numbers",
+    snippet = [=[
+function()
+    -- Change tooltip1 to your value
+    -- Change [1] to other your trigger number
+    -- If using a tooltip value, be sure to tick Use Tooltip Values in the trigger
+    if aura_env.states[1] and aura_env.states[1].tooltip1 then
+        return AbbreviateNumbers(aura_env.states[1].tooltip1)
+    end
+end]=]
+  },
+  {
+    name = "Text: Colored Name",
+    snippet = [=[
+function()
+    if aura_env.states[1] and aura_env.states[1].unit then
+        return WA_ClassColorName(aura_env.states[1].unit)
+    end
+end]=]
+  },
+  {
+    name = "Utility: Dump",
+    snippet = IsAddOnLoaded("ViragDevTool") and [[
```

Comment:
This seeems to check if ViragDevTool is loaded while loading WeakAuras, meaning the contents of the snippet depend on the order of the addon loading?

Reply:
Good catch. This was a late addition and I thought I was so clever. 
It would be nice to have a Dump snippet in there but the Blizzard one (adding the DevTools addon load) and the VDT one (supporting an outside addon) are both awkward. 

I just take this out for now and can always add later if a better idea is generated.

Reply:
the better idea would be to allow "snippet" to be a function which creates a string.

Then that function can check, and return the right value. (Obviously this needs some changes in those places where snippet is used.)